add cancel scenarios

georgescharlesbrain opened the pull request:
The additional code provides the following new cancel scenarios which prevent player fund lock-up:

*   a player or judge can cancel a bet during the funding phase when one of the players chickens out
*   players can unanimously decide to cancel a bet during the voting phase (e.g. judges don't show up)

Some of the phase numbers in bet.tests.move are replaced by the actual phase name.  
New cancel unit tests were added.

'sui move test' runs successfully.

georgescharlesbrain committed:
add more cancel scenarios

juzybits commented:
This is cool, thank you for the PR! I have a busy week ahead but I will try to find the time to review your changes soon.

juzybits committed:
[sui] tweak PR: 'add more cancel scenarios'

~ remove trailing spaces
~ sort .gitignore lines
~ bet.move:
  - move error constants to the top
  - transform `){` into `) {`
  - add `;` after `if` statements
  - cancel():
    · simplify logic
    · simplify exceptions
    · assert bet.phase early on
~ bet.tests.move:
  order test_cancel_*() so that success cases go first and error cases go last
  tweak function names

juzybits committed:
[sui] increment version to 2.0.0
